ci-operator/step-registry/baremetalds/e2e/ovn/debug/console-gather/baremetalds-e2e-ovn-debug-console-gather-commands.sh

```diff
@@ -0,0 +1,167 @@
+}
+
+# Gather logs regardless of what happens after this
+trap getlogs EXIT
+
+# shellcheck disable=SC2087
+ssh "${SSHOPTS[@]}" "root@${IP}" bash - <<EOF
+
+source ~/dev-scripts-additional-config
+
+firstNodeNotReady=\$(oc get nodes -o json | jq -r '[.items[] | select(([.status.conditions[] | select(.type=="Ready") | select(.status!="True")] | length)==1) | .status.nodeInfo.systemUUID][0] // empty')
+if [ -z "\$firstNodeNotReady" ]; then
+  echo "All nodes are healthy, nothing to do"
+  exit 0
+fi
+
+output=console-gather.\${firstNodeNotReady}.tar.xz
+
+echo "### Gathering logs for node \$firstNodeNotReady"
+dnf install -y expect
+mkdir ${tmpdir}
+expect -c "
+    set prompt {][$]}
+    set return \"\r\"
+    set disconnect \"\x1d\"
+
+    spawn virsh console --force \$firstNodeNotReady
+    match_max -i \\\$spawn_id 3000000
+    expect_before {
+      timeout { exit 1 }
+    }
+
+    set timeout 30
+
+    # login
+    expect -re {Escape character.*}
+    send \\\$return
+    expect -re {.*login: \$}
+    send {core}
+    send \\\$return
+    expect {Password:}
+    send {${PASSWD}}
+    send \\\$return
+    expect -re \\\$prompt
+
+    # create temp dir
+    send {mkdir ${tmpdir}}
+    send \\\$return
+    expect -re \\\$prompt
+    send {cd ${tmpdir}}
+    send \\\$return
+    expect -re \\\$prompt
+    send {chmod a+rw .}
+    send \\\$return
+    expect -re \\\$prompt
+
+    # collect information
+
+    # sos report
+    set timeout 600
+    send {toolbox sos report --batch -q -o logs,networking,networkmanager,conntrack,openvswitch,frr,crio -k crio.logs=on --tmp-dir ${tmpdir}}
+    send \\\$return
+    expect -re \\\$prompt
+    set timeout 30
+
+    # get frr pod id
+    send {frr=\\\$(sudo crictl ps --label io.kubernetes.container.name=frr -o json | jq -r '.[] | .[0] | .id')}
+    send \\\$return
+    expect -re \\\$prompt
+
+    # get frr info
+    send {sudo crictl exec \\\$frr vtysh -c 'show running-conf' > ${tmpdir}/frr.ip_route_vrf_all.\${firstNodeNotReady}.txt}
+    send \\\$return
+    expect -re \\\$prompt
+    send {sudo crictl exec \\\$frr vtysh -c 'show ip route vrf all' > ${tmpdir}/frr.ip_route_vrf_all.\${firstNodeNotReady}.txt}
+    send \\\$return
+    expect -re \\\$prompt
+    send {sudo crictl exec \\\$frr vtysh -c 'show ip bgp vrf all ipv4 summary' > ${tmpdir}/frr.ip_bgp_vrf_all_ipv4_summary.\${firstNodeNotReady}.txt}
+    send \\\$return
+    expect -re \\\$prompt
+    send {sudo crictl exec \\\$frr vtysh -c 'show bgp vrf all ipv4' > ${tmpdir}/frr.bgp_vrf_all_ipv4.\${firstNodeNotReady}.txt}
+    send \\\$return
+    expect -re \\\$prompt
+    send {sudo crictl exec \\\$frr vtysh -c 'show bgp vrf all ipv4 neighbor' > ${tmpdir}/frr.bgp_vrf_all_ipv4_neighbor.\${firstNodeNotReady}.txt}
+    send \\\$return
+    expect -re \\\$prompt
+    send {sudo crictl exec \\\$frr vtysh -c 'show bfd vrf all peer' > ${tmpdir}/frr.bfd_vrf_all_peer.\${firstNodeNotReady}.txt}
+    send \\\$return
+    expect -re \\\$prompt
+
```

Comment:
Shall we also try to collect the log of ovn-k pods?

Reply:
That's already included in the sosreport being collected